cypress/support/commands.ts

```diff
@@ -1,37 +1,59 @@
 /// <reference types="cypress" />
 // ***********************************************
-// This example commands.ts shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-//
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
+
+// Keeps requesting a URL until it contains and XML <item>
+function waitForRssItems(url, title, retries = 0) {
+  cy.request({ url: url, failOnStatusCode: false }).then((resp) => {
+    if (resp.status === 200) {
+      const parser = new DOMParser();
+
+      const xml = resp.body;
+      const doc = parser.parseFromString(xml, "application/xml") as XMLDocument;
+
+      const items = doc.querySelectorAll("item");
+
+      if (items.length) {
+        const episode = Array.from(items)[0];
+        const episodeTitle = episode
+          .getElementsByTagName("title")[0]
+          .textContent.trim();
+
+        const enclosure = episode.getElementsByTagName("enclosure")[0];
+        const enclosureUrl = enclosure.getAttribute("url");
+        const enclosureLength = enclosure.getAttribute("length");
+
+        if (title === episodeTitle) {
+          expect(title).to.equal(episodeTitle);
+          expect(enclosureLength).to.equal("40557");
+        }
+
+        cy.request({
+          url: enclosureUrl,
+          encoding: "binary",
+        }).then((response) => {
+          cy.writeFile("test.mp3", response.body, "binary");
+          cy.readFile("test.mp3", "base64").then((mp3) => {
+            expect(mp3.length).to.equal(54076);
```

Comment:
Is this the length of the original file?  Or did you just figure out what the cdn-arranger results in, after it does all the strip-then-add-id3?

Reply:
I ran the test with a `1` and then changed it to the actual value it failed on, so it is based on the actual file coming out of Dovetail, but beyond that I don't know how much it's taking into account.